src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -171,122 +184,371 @@ final case class ThermalGrid(
       qDot,
     )
 
-  /** Handles the case, when a grid has infeed. First, heat up all the houses to
-    * their maximum temperature, then fill up the storages
+  /** Handles the case, when a grid has infeed. Depending which entity has some
+    * heat demand the house or the storage will be heated up / filled up.
     *
     * @param relevantData
     *   data of heat pump including state of the heat pump
     * @param lastAmbientTemperature
     *   Ambient temperature valid up until (not including) the current tick
-    * @param state
-    *   Current state of the houses
+    * @param thermalGridState
+    *   Current state of the thermal Grid
+    * @param isRunning
+    *   determines whether the heat pump is running or not
```

Comment:
If this refers to the last state, maybe call this `wasRunning`?

Reply:
No, this refers to the actual moment. So if the Hp is running from this tick on.